### DISCUSSION TITLE
Some questions about gpus.

### DISCUSSION THREAD

**Discussion by Liujiaqi-jlu** [gem5/gem5#1730] [Q&A]
Hello, I have some questions about GPUs. I noticed that in the gem5 source code, there's an "amdgpu" folder and a "gpu-compute" folder. I'd like to know the difference between these two GPU-related folders. I saw a GPU training video a few months ago that used "amdgpu". So, what is the purpose of the "gpu-compute" folder?

I'm also interested in creating a simple example involving multi-GPU interconnection. Could you provide some guidance on this? Thanks!

**Comment by abmerop** [comment]
The `gpu-compute` folder contains code for the shader pipeline model. There are two amdgpu folders where `arch/amdgpu` contains the ISA definitions and `dev/amdgpu` contains all of the uncore components of the GPU (non-shader SoC hardware models).

I am not sure about multiple GPUs.  I have heard of some success using SE mode but it is not in upstream gem5.  In next year's release probably stdlib will support multiple GPUs.
